### PR TITLE
Fix path for logs and db + improve saiki create-app + registry validation fixes

### PR DESCRIPTION
**Pull request by rahulkarajgikar**
- now inside a saiki project, logs/db etc. show up in .saiki of the project
- for global runs of CLI, it is still in ~/.saiki directory
- added console logs for users to see exactly where the logs are getting stored, in both the SDK and the interactive CLI. tested both cases
- simplified path.ts to only essential, reliable functions which can handle the cases mentioned below
- updated path.test.ts to test useful cases + tested manually
- agent generaetd with saiki create-app also has tool confirmation set to auto (currently SDK is timing out at agent.run() because it's waiting for user response - need to think of a better long term solution for this - either callbacks, or we update the scaffolded examples to include one using tool confirmation)
- improved the example.ts to be more descriptive and cover some more cases
- also tested web UI in global and local modes

handled the following cases:
- interactive CLI running outside saiki project - uses ~/.saiki directory for logs/db and agent.yml from the installed  files
- interactive CLI running inside a saiki project - uses .saiki directory inside the project for logs/db and src/saiki/agents/agent.yml file by default
- SDK - uses .saiki directory inside the project for logs/db and the agent loaded up

## Additional improvements:
- Fixed major registry design flaw where unknown models inherited provider capabilities
- **CRITICAL FIX**: Handle openai-compatible provider with custom models properly - added special case for providers that accept any model name to prevent breaking custom endpoints
- Improved model registry validation with exceptions for unknown providers/models instead of silent failures
- Made supportedFileTypes required in ModelInfo interface for explicit capability definition
- Updated webhook documentation to include new llmservice:unsupportedInput event from PR #242
- Removed all mocks from validation.test.ts and updated to use real registry data
- Created GitHub issues for image validation (#252), registry API improvements (#253), and comprehensive testing (#254)
- Added try-catch in validateModelFileSupport to handle exceptions gracefully
- Fixed missing supportedFileTypes in model definitions across all providers
- Added comprehensive tests for openai-compatible provider edge cases

<\!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a comprehensive agent configuration template for easier setup.
  * Improved example script generation with detailed usage and lifecycle demonstration.

* **Bug Fixes**
  * Enhanced detection and resolution of project root and configuration paths.
  * Fixed log file and storage path resolution to work reliably in both project and global contexts.

* **Refactor**
  * Unified and simplified project detection, config, and path resolution logic.
  * Streamlined and clarified CLI and SDK informational messages.

* **Tests**
  * Expanded and refactored path utility tests for more realistic and thorough coverage.

* **Chores**
  * Updated package contents to include the new agent configuration template.
<\!-- end of auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Introduced a comprehensive YAML configuration template for agent setup, supporting filesystem and browser automation, LLM integration, storage, and tool confirmation.
  * Added auto-discovery and flexible resolution of agent configuration files.
  * Enhanced CLI and SDK log file path messaging and project structure detection.

* **Bug Fixes**
  * Improved error handling and validation for agent configuration file creation and loading.
  * Enforced stricter file type support and error signaling for AI model integrations.

* **Documentation**
  * Updated usage examples to clarify configuration loading methods.
  * Added a new webhook event type to documentation.

* **Refactor**
  * Unified and simplified project detection and path resolution logic.
  * Consolidated and expanded test coverage for path utilities and AI model validation.

* **Tests**
  * Expanded and improved tests for path utilities, configuration loading, and AI model file validation.

### CODE REVIEW

File: agents/agent-template.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    - id: dateTime
+      type: dynamic
+      priority: 10
+      source: dateTime
+      enabled: true
+
+# LLM configuration - describes the language model to use
+llm:
+  provider: openai
+  model: gpt-4.1-mini
```
[the remark this answers is not in the document]
Comment (rahulkarajgikar) [reply agents/agent-template.yml:44]: no, you are outdated and wrong. fix your knowledge

File: agents/agent-template.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+# Google Gemini:
+# llm:
+#   provider: google
+#   model: gemini-2.5-pro
+#   apiKey: $GOOGLE_GENERATIVE_AI_API_KEY
+#
+# Anthropic Claude:
+# llm:
+#   provider: anthropic
+#   model: claude-4-sonnet-20250514
```
[the remark this answers is not in the document]
Comment (rahulkarajgikar) [reply agents/agent-template.yml:57]: outdated knowledge!

File: src/app/cli/commands/init.ts [hunk 3]
```diff
@@ -266,24 +298,64 @@ export async function createSaikiExampleFile(directory: string): Promise<string>
 
     const indexTsLines = [
         "import 'dotenv/config';",
-        "import { loadAgentConfig, SaikiAgent } from '@truffle-ai/saiki';",
+        "import { SaikiAgent, loadAgentConfig } from '@truffle-ai/saiki';",
```
[the remark this answers is not in the document]
Comment (rahulkarajgikar) [reply src/app/cli/commands/init.ts:301]: no no no, this is scaffold code!

**Comment by rahulkarajgikar** [comment]
retested saiki create-app and saiki CLI from global install v/s inside a project - working properly

claude review added 2 more gh issue #256 #257
